Source/Core/Resource.js

```diff
@@ -336,6 +339,14 @@ define([
         this.retryAttempts = defaultValue(options.retryAttempts, 0);
         this._retryCount = 0;
 
+        /**
+         * Whether to vertically flip the image during fetch and decode. Only applies when requesting
+         * an image and the browser supports createImageBitmap.
+         *
+         * @type {Boolean}
+         */
+        this.flipImage = defaultValue(options.flipImage, true);
```

Comment:
This feels like a leaky abstraction, I'm also not sure why it would go on the Resource constructor and not just on `fetchImage` (since that's the only place it applies, right?

Reply:
I also couldn't find from your PR description what this option is used for and why.  Isn't it a completely internal implementation detail?  Couldn't we always/never flip based on what we detect regarding createImageBitmap?  Why does the calling code need control over this?

Reply:
Sorry for not making my justification here clearer.

We can't make decisions about whether to flip inside `Resource` because whether we flip or not depends on what we're going to do with this image. All textures in CesiumJS are flipped by default during texture upload. This seems to be because imagery expects this orientation.

Models with textures will disable this flipping to get the correct orientation. This means that there's no one correct orientation. If we always flip, models will be upside down. If we never flip, imagery will be upside down. I spent some time trying to see if I could get all imagery to be flipped in the shader so we can just simply never have to worry about flipping during texture upload but I couldn't get it to work correctly with the projection. 

As for why it's not just on `fetchImage`, you mean so that it would be like this?

```
Resource.prototype.fetchImage = function (preferBlob, flipY) {
```

The reason I implemented it the way it is right now is to get it to work for this case:

https://github.com/AnalyticalGraphicsInc/cesium/blob/e31eae334a4f8ddafe520ff59c0d2bf2ba16a33f/Source/Core/Resource.js#L899-L904

Where the local file-scope `fetchImage` function only takes a `resource` as a parameter. But I could just pass the flag there too. I do agree it's weird to have an option on `Resource` that only applies during a specific function call.

Reply:
Thanks for the details. Yes, I would just have fetchImage take an options object with `preferBlob` and `flipY` propreties (should be an easy deprecation since you can check if a bool was previously being passed).